packages/frontend/src/hooks/useKeycard.ts

```diff
@@ -1,27 +1,80 @@
+import { assert } from "@std/assert";
 import { useLocalStorage } from "@uidotdev/usehooks";
 import { generatePrivateKey, privateKeyToAccount } from "viem/accounts";
+import { getLogger } from "@logtape/logtape";
+import { useWalletClient } from "wagmi";
+import { useQuery } from "@tanstack/react-query";
+
+import { RelayClient } from "@massmarket/client";
+import { getWindowLocation } from "@massmarket/utils";
 
 import { useShopId } from "./useShopId.ts";
 import { KeycardRole } from "../types.ts";
+import { useRelayEndpoint } from "./useRelayEndpoint.ts";
```

Comment:
this is going to cause problem. We are also going to need to change this over to tanstack's useQuery. For now maybe do something awful and just create a function in here or seperate file that return the releyEndpoint infromation a promise. We then will call and await it in the body of the below useQuery.